CMakeLists.txt

```diff
@@ -264,114 +440,130 @@ install(
     COMPONENT headers
 )
 
-include(ExternalProject)
-
-# include(mergestaticlibs)
 # ------------------------------------------------------------
 # Enable OpenMP support?
 # ------------------------------------------------------------
-if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-    option(OPENMP_ENABLE "Enable openMP support" OFF)
-else()
-    option(OPENMP_ENABLE "Enable openMP support" ON)
-endif()
-
-if(OPENMP_ENABLE)
-    # message(STATUS "****** finding OpenMP support")
-    find_package(OpenMP)
+if(ENABLE_OPENMP)
+    find_package(OpenMP QUIET)
     if(OPENMP_FOUND)
-        if(NOT ${CMAKE_VERSION} VERSION_LESS "3.9")
-            link_libraries(OpenMP::OpenMP_C OpenMP::OpenMP_CXX)
-        else()
-            add_compile_options(
-                $<$<COMPILE_LANGUAGE:C>:${OpenMP_C_FLAGS}>
-                $<$<COMPILE_LANGUAGE:CXX>:${OpenMP_CXX_FLAGS}>
-            )
-            link_libraries(${OpenMP_C_FLAGS})
-        endif()
+        # TODO this needs to apply to each thing that might want openmp
+        # TODO also the COMPILE_LANGUAGE generator was in cmake as of 3.3, why was it conditional on 3.9?
+        target_compile_options(griddyn_base INTERFACE
+            $<$<COMPILE_LANGUAGE:C>:${OpenMP_C_FLAGS}>
+            $<$<COMPILE_LANGUAGE:CXX>:${OpenMP_CXX_FLAGS}>
+        )
+        # OpenMP::OpenMP_CXX wasn't added until CMake 3.9
+        target_link_libraries(griddyn_base INTERFACE ${OpenMP_C_FLAGS})
     else(OPENMP_FOUND)
         message(
             STATUS
                 "Disabling OpenMP support, could not determine compiler flags"
         )
-        set(OPENMP_ENABLE FALSE)
+        set(ENABLE_OPENMP OFF)
     endif(OPENMP_FOUND)
-else(OPENMP_ENABLE)
-    set(OPENMP_FOUND FALSE)
-endif(OPENMP_ENABLE)
+endif(ENABLE_OPENMP)
 
-if(OPENMP_FOUND)
-    option(SUNDIALS_OPENMP "Enable SUNDIALS NVector openMP implementation" ON)
-    option(GRIDDYN_OPENMP "Enable openmp internal to GridDyn--not used yet" OFF)
-endif(OPENMP_FOUND)
+cmake_dependent_option(
+    ENABLE_OPENMP_SUNDIALS
+    "Enable SUNDIALS NVector openMP implementation"
+    ON
+    "ENABLE_OPENMP"
+    OFF
+)
 
-option(DISABLE_MULTITHREADING "disable multithreading in GridDyn libraries" OFF)
+cmake_dependent_option(
+    # TODO look at this
+    ENABLE_OPENMP_GRIDDYN
+    "Enable openmp internal to GridDyn--not used yet"
+    OFF
+    "ENABLE_OPENMP"
+    OFF
+)
 
 # ------------------------------------------------------------
 # Find multithreading headers and includes
 # ------------------------------------------------------------
-
-if(NOT DISABLE_MULTITHREADING)
-    if(NOT WIN32)
-        set(THREADS_PREFER_PTHREAD_FLAG ON)
-    elseif(MSYS)
+if(ENABLE_MULTITHREADING)
+    if(NOT WIN32 OR MSYS)
+        # TODO ?? shouldn't this just be an if (MSYS)? anything else'll use pthreads and win32 probably won't have it
```

Comment:
the standalone mingw compiler has some weird stuff that interacts oddly, so it sometimes found pthreads when it shouldn't have.  We could probably write this condition differently but If it is on Windows it should not use PTHREAD unless it is running in MSYS.